src/lsp/cobol_parser/parser_outputs.ml

```diff
@@ -0,0 +1,32 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                        SuperBOL OSS Studio                             *)
+(*                                                                        *)
+(*  Copyright (c) 2022-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(* All rights reserved.                                                   *)
+(* This source code is licensed under the GNU Affero General Public       *)
+(* License version 3 found in the LICENSE.md file in the root directory   *)
+(* of this source tree.                                                   *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Gathers some types used to define outputs for the parser engine. *)
+(* We are only defining types here so an MLI would only be redundant. *)
+
+open Cobol_common.Srcloc.TYPES
+
+type tokens_with_locs = Grammar_tokens.token with_loc list
+
+type artifacts =
+  {
+    tokens: tokens_with_locs Lazy.t;
+    pplog: Cobol_preproc.log;
+    comments: Cobol_preproc.comments;
+  }
+
+type ('a, 'm) output =
```

Comment:
* I cannot see the use of the `'m` parameter. 
* I don't understand what `'a` and `'m` stand for, maybe using longer names that 1-character letters would help
* I don't see the point of using yet more GADTs. I think we should go the other way around: progressively getting rid of GADTs, because they make the code impossible to understand for anybody else than their author.

Reply:
Ok I'll document that aspect.